massa-pool-worker/src/endorsement_pool.rs

```diff
@@ -72,6 +80,12 @@ impl EndorsementPool {
             }
         }
         self.storage.drop_endorsement_refs(&removed);
+
+        // Send to Denunciation factory
+        let de_interest = DenunciationInterest::Final(self.last_cs_final_periods.clone());
+        self.de_items_tx
+            .send(de_interest)
```

Comment:
I think we can use try_send to avoid blocking here

Reply:
Fixed